[2025-03-08T23:57:39.399+0000] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2025-03-08T23:57:39.426+0000] {taskinstance.py:2612} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: career_analytics_pipeline.load_data manual__2025-03-08T23:57:27.622532+00:00 [queued]>
[2025-03-08T23:57:39.444+0000] {taskinstance.py:2612} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: career_analytics_pipeline.load_data manual__2025-03-08T23:57:27.622532+00:00 [queued]>
[2025-03-08T23:57:39.458+0000] {taskinstance.py:2865} INFO - Starting attempt 1 of 2
[2025-03-08T23:57:39.482+0000] {taskinstance.py:2888} INFO - Executing <Task(PythonOperator): load_data> on 2025-03-08 23:57:27.622532+00:00
[2025-03-08T23:57:39.492+0000] {standard_task_runner.py:72} INFO - Started process 190 to run task
[2025-03-08T23:57:39.497+0000] {standard_task_runner.py:104} INFO - Running: ['***', 'tasks', 'run', 'career_analytics_pipeline', 'load_data', 'manual__2025-03-08T23:57:27.622532+00:00', '--job-id', '40', '--raw', '--subdir', 'DAGS_FOLDER/Careers_Data_Pipeline.py', '--cfg-path', '/tmp/tmpnd0rm7ii']
[2025-03-08T23:57:39.503+0000] {standard_task_runner.py:105} INFO - Job 40: Subtask load_data
[2025-03-08T23:57:39.574+0000] {task_command.py:467} INFO - Running <TaskInstance: career_analytics_pipeline.load_data manual__2025-03-08T23:57:27.622532+00:00 [running]> on host b128859cb372
[2025-03-08T23:57:39.877+0000] {taskinstance.py:3131} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='career_analytics_pipeline' AIRFLOW_CTX_TASK_ID='load_data' AIRFLOW_CTX_EXECUTION_DATE='2025-03-08T23:57:27.622532+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-03-08T23:57:27.622532+00:00'
[2025-03-08T23:57:39.880+0000] {taskinstance.py:731} INFO - ::endgroup::
[2025-03-08T23:57:39.940+0000] {warnings.py:110} WARNING - /opt/***/dags/Careers_Data_Pipeline.py:144: FutureWarning: Passing literal json to 'read_json' is deprecated and will be removed in a future version. To read from a literal string, wrap it in a 'StringIO' object.
  professionals_df = pd.read_json(ti.xcom_pull(key='professionals_df', task_ids='transform_data'))

[2025-03-08T23:57:39.962+0000] {warnings.py:110} WARNING - /opt/***/dags/Careers_Data_Pipeline.py:145: FutureWarning: Passing literal json to 'read_json' is deprecated and will be removed in a future version. To read from a literal string, wrap it in a 'StringIO' object.
  jobs_df = pd.read_json(ti.xcom_pull(key='jobs_df', task_ids='transform_data'))

[2025-03-08T23:57:39.982+0000] {warnings.py:110} WARNING - /opt/***/dags/Careers_Data_Pipeline.py:146: FutureWarning: Passing literal json to 'read_json' is deprecated and will be removed in a future version. To read from a literal string, wrap it in a 'StringIO' object.
  skills_df = pd.read_json(ti.xcom_pull(key='skills_df', task_ids='transform_data'))

[2025-03-08T23:57:40.005+0000] {warnings.py:110} WARNING - /opt/***/dags/Careers_Data_Pipeline.py:147: FutureWarning: Passing literal json to 'read_json' is deprecated and will be removed in a future version. To read from a literal string, wrap it in a 'StringIO' object.
  certifications_df = pd.read_json(ti.xcom_pull(key='certifications_df', task_ids='transform_data'))

[2025-03-08T23:57:40.024+0000] {warnings.py:110} WARNING - /opt/***/dags/Careers_Data_Pipeline.py:148: FutureWarning: Passing literal json to 'read_json' is deprecated and will be removed in a future version. To read from a literal string, wrap it in a 'StringIO' object.
  educations_df = pd.read_json(ti.xcom_pull(key='educations_df', task_ids='transform_data'))

[2025-03-08T23:57:40.130+0000] {taskinstance.py:3310} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/models/taskinstance.py", line 767, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/models/taskinstance.py", line 733, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/models/baseoperator.py", line 406, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/operators/python.py", line 238, in execute
    return_value = self.execute_callable()
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/operators/python.py", line 256, in execute_callable
    return runner.run(*self.op_args, **self.op_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/airflow/dags/Careers_Data_Pipeline.py", line 191, in load_data_task
    con.execute("""
duckdb.duckdb.BinderException: Binder Error: Referenced column "title" not found in FROM clause!
Candidate bindings: "role", "industry", "start_date"

LINE 6:             title,
                    ^
[2025-03-08T23:57:40.150+0000] {taskinstance.py:1225} INFO - Marking task as UP_FOR_RETRY. dag_id=career_analytics_pipeline, task_id=load_data, run_id=manual__2025-03-08T23:57:27.622532+00:00, execution_date=20250308T235727, start_date=20250308T235739, end_date=20250308T235740
[2025-03-08T23:57:40.172+0000] {taskinstance.py:340} INFO - ::group::Post task execution logs
[2025-03-08T23:57:40.175+0000] {standard_task_runner.py:124} ERROR - Failed to execute job 40 for task load_data (Binder Error: Referenced column "title" not found in FROM clause!
Candidate bindings: "role", "industry", "start_date"

LINE 6:             title,
                    ^; 190)
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/task/task_runner/standard_task_runner.py", line 117, in _start_by_fork
    ret = args.func(args, dag=self.dag)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/cli/cli_config.py", line 49, in command
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/utils/cli.py", line 115, in wrapper
    return f(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/cli/commands/task_command.py", line 483, in task_run
    task_return_code = _run_task_by_selected_method(args, _dag, ti)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/cli/commands/task_command.py", line 256, in _run_task_by_selected_method
    return _run_raw_task(args, ti)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/cli/commands/task_command.py", line 341, in _run_raw_task
    return ti._run_raw_task(
           ^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/utils/session.py", line 97, in wrapper
    return func(*args, session=session, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/models/taskinstance.py", line 3004, in _run_raw_task
    return _run_raw_task(
           ^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/models/taskinstance.py", line 273, in _run_raw_task
    TaskInstance._execute_task_with_callbacks(
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/models/taskinstance.py", line 3158, in _execute_task_with_callbacks
    result = self._execute_task(context, task_orig)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/models/taskinstance.py", line 3182, in _execute_task
    return _execute_task(self, context, task_orig)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/models/taskinstance.py", line 767, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/models/taskinstance.py", line 733, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/models/baseoperator.py", line 406, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/operators/python.py", line 238, in execute
    return_value = self.execute_callable()
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/operators/python.py", line 256, in execute_callable
    return runner.run(*self.op_args, **self.op_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/airflow/dags/Careers_Data_Pipeline.py", line 191, in load_data_task
    con.execute("""
duckdb.duckdb.BinderException: Binder Error: Referenced column "title" not found in FROM clause!
Candidate bindings: "role", "industry", "start_date"

LINE 6:             title,
                    ^
[2025-03-08T23:57:40.237+0000] {local_task_job_runner.py:266} INFO - Task exited with return code 1
[2025-03-08T23:57:40.262+0000] {taskinstance.py:3900} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2025-03-08T23:57:40.277+0000] {local_task_job_runner.py:245} INFO - ::endgroup::
